### PR TITLE
[ASLayoutElementStyle] implement -description with sizing info

### PR DESCRIPTION
**Pull request by hannahmbanana**
image

### CODE REVIEW

File: AsyncDisplayKit/Private/ASEnvironmentInternal.mm [hunk 1]
```diff
@@ -153,6 +153,9 @@ ASEnvironmentState ASEnvironmentMergeObjectAndState(ASEnvironmentState environme
     if (parentLayoutOptionsState.ascender == defaultState.ascender) {
       parentLayoutOptionsState.ascender = layoutOptionsState.ascender;
     }
+    if (parentLayoutOptionsState.descender == defaultState.descender) {
+      parentLayoutOptionsState.descender = layoutOptionsState.descender;
+    }
```
Comment (hannahmbanana) [inline AsyncDisplayKit/Private/ASEnvironmentInternal.mm]: @maicki - this wasn't left out on purpose was it?

**Comment by hannahmbanana** [comment]
@Adlai-Holler - ready for final review.

**Comment by hannahmbanana** [comment]
^ rebase to see if that resolves the Jenkins build failure

**Comment by hannahmbanana** [comment]
^ this passed running the build script locally :(

File: AsyncDisplayKit.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1956,12 +1956,15 @@
 			isa = PBXNativeTarget;
 			buildConfigurationList = 058D09D2195D04C000B7D73C /* Build configuration list for PBXNativeTarget "AsyncDisplayKitTests" */;
 			buildPhases = (
+				E2F06B8443CED83576ECA681 /* 📦 Check Pods Manifest.lock */,
```
Comment (Adlai-Holler) [inline AsyncDisplayKit.xcodeproj/project.pbxproj]: Let's get these emoji out of here

File: AsyncDisplayKit/Layout/ASLayoutElement.mm [hunk 3]
```diff
@@ -437,6 +437,77 @@ - (CGPoint)layoutPosition
...
+  if (self.alignSelf != defaultState.alignSelf) {
+    [result addObject:@{ @"alignSelf" : [@[@"ASStackLayoutAlignSelfAuto",
+                                          @"ASStackLayoutAlignSelfStart",
+                                          @"ASStackLayoutAlignSelfEnd",
+                                          @"ASStackLayoutAlignSelfCenter",
+                                          @"ASStackLayoutAlignSelfStretch"] objectAtIndex:self.alignSelf] }];
+  }
+  
+  if (self.ascender != defaultState.ascender) {
+    [result addObject:@{ @"ascender" : [NSNumber numberWithFloat:self.ascender] }];
```
Comment (Adlai-Holler) [inline AsyncDisplayKit/Layout/ASLayoutElement.mm]: Not critical, but let's prefer the equivalent `@(self.ascender)`

File: AsyncDisplayKit/Layout/ASLayoutElement.mm [hunk 4]
```diff
@@ -437,6 +437,77 @@ - (CGPoint)layoutPosition
...
+  }
+  
+  const ASEnvironmentLayoutOptionsState defaultState = ASEnvironmentLayoutOptionsStateMakeDefault();
+  
+  if (self.alignSelf != defaultState.alignSelf) {
+    [result addObject:@{ @"alignSelf" : [@[@"ASStackLayoutAlignSelfAuto",
+                                          @"ASStackLayoutAlignSelfStart",
+                                          @"ASStackLayoutAlignSelfEnd",
+                                          @"ASStackLayoutAlignSelfCenter",
+                                          @"ASStackLayoutAlignSelfStretch"] objectAtIndex:self.alignSelf] }];
```
Comment (Adlai-Holler) [inline AsyncDisplayKit/Layout/ASLayoutElement.mm]: Ha! How 'bout that!

**Comment by Adlai-Holler** [comment]
Hmm running the tests is still failing locally for me (Xcode 8 iPhone 7 simulator). I'll investigate later today unless you get to it first @hannahmbanana

**Comment by hannahmbanana** [comment]
I've run out of ideas on how to fix this. I'm going to try opening a new PR and see if that magically works.

**Comment by hannahmbanana** [comment]
@Adlai-Holler - I'm getting the duplicate protocol definition error again.

**Comment by Adlai-Holler** [comment]
Superseded by #2495

**Pull request closed without merging**